src/unaryops.cu

```diff
@@ -80,6 +82,52 @@ struct UnaryOp {
 };
 
 
+template<typename T, typename Tout, typename F>
+struct UnaryOpArrow {
```

Comment:
I wonder if this should replace UnaryOp, but I suppose this is fine for now and something we can ask them about.